apps/authentication-service/src/shared/auth/authService.test.ts

```diff
@@ -0,0 +1,74 @@
+import fs from "fs";
+import { http, HttpResponse } from "msw";
+
+import * as tokenUtil from "@/shared/tokenUtils";
+import { tokenFile } from "@/shared/tokenUtils";
+import { AUTH0_AUTH_TOKEN } from "@/test/handlers";
+import { server } from "@/test/testServer";
+import { getAccessToken } from "./authService";
+import { exampleToken } from "@/test/testData/users";
+
+const exampleErrorString = "Failed to authenticate using M2M Credentials";
+
+describe("authService", () => {
+  beforeAll(() => {
+    jest.spyOn(tokenUtil, "getLocalToken").mockReturnValue(exampleToken);
+    jest.spyOn(tokenUtil, "setLocalToken").mockImplementation(() => {});
+  });
+
+  beforeEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  describe("getAccessToken", () => {
+    it("handles an error from the token endpoint", async () => {
+      server.use(
+        http.post(
+          AUTH0_AUTH_TOKEN,
+          () =>
+            new HttpResponse(null, {
+              status: 500,
+            }),
+        ),
+      );
+      jest.spyOn(fs, "existsSync").mockReturnValue(false);
+      jest.spyOn(tokenUtil, "getLocalToken").mockReturnValue(undefined);
+
+      await expect(getAccessToken()).rejects.toEqual(
+        new Error(exampleErrorString),
+      );
+    });
+
+    it("fetches a new access token if there's no token, and sets it in the cache", async () => {
+      jest.spyOn(fs, "writeFileSync");
+      jest.spyOn(tokenUtil, "getLocalToken").mockReturnValueOnce(undefined);
+      jest.spyOn(tokenUtil, "getCachedToken").mockReturnValue(undefined);
+
+      await getAccessToken();
```

Comment:
We should be testing that the correct token was returned.

Reply:
It would be good to mock this to return a specific token, so we're sure that the token didn't come from the cache. It came from fetching a new one.